platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,346 @@
+cannot happen until that roster has enough nodes able to recover a threshold number of shares so that an aggregation of
+node signatures can generate the ledger signature.
+
+##### Elliptic Curve Decisions
+
+Our first implementation of Groth21 will use the ALT_BN128 elliptic curve which is in use and verifiable by EVMs.  
+If and when the Ethereum ecosystem adopts BLS12_381, we will likely switch over to that more secure curve.
+
+##### New Elliptic Curve Node Keys
+
+Each node will need a new long-term EC key pair in addition to the existing RSA key pair. The EC key pair will be
+used in the Groth21 algorithm to generate and decrypt `TSS-Messages`. These new EC Keys will not be used for
+signing messages, only for generating the shares. It is the share keys that are used to sign messages. The public
+keys of these long-term node specific EC keys must be in the address book.
+
+##### Groth21 Drawbacks
+
+The Groth21 algorithm is not able to model arbitrary precision proportions of weight assigned to nodes in the
+network. For example if each node in a 4 node network received 1 share, then every share has a weight of 1/4.
+If there are a total of N shares, then the distribution of weight can only be modeled in increments of (1/N) * total
+weight. The cost of re-keying the network in an address book change is quadratic in the number of shares. This forces us
+to pick a number of total shares with a max value in the thousands. This modeling of weight is a discrete using small
+integer precision.
+
+#### Alternatives Considered
+
+The list of options considered here were based off of prototypes developed by Rohit Sinha, the Swirlds Labs
+cryptography expert. The Groth21 algorithm was chosen because it was efficient for use in smart contract
+verification, and we could assign a multiplicity of shares to nodes to get close enough in modeling the
+distribution of weight between nodes.
+
+| Requirement | hinTS | Groth21 |
+|-------------|-------|---------|
+| TSS-001     | Yes   | Yes     |
+| TSS-002     | Yes   | Yes     |
+| TSS-003     | Yes   | Yes     |
+| TSS-004     | Yes   | Yes     |
+| TSS-005     | Yes   | Yes     |
+| TSS-006     | Yes   | Yes     |
+| TSS-007     | Yes   | Yes     |
+| TSS-008     | No    | Yes     |
+| TSS-009     | Yes   | Yes     |
+| TSS-010     | Yes   | No      |
+
+##### hinTS
+
+The hinTS algorithm is a threshold signature scheme that is able to measure double precision distributions of weight
+across nodes. The complicating factor with hinTS is that during an address book change over, a recursive proof needs
+to be constructed to prove that the new roster is a descendent of the original genesis roster. Validation of this
+recursive proof proved too expensive for EVM smart contracts.
+
+### Goals
+
+The following capabilities are the goals of this proposal:
+
+1. `Ledger Signing API` - Given a message, the consensus nodes sign the message with shares and aggregate the signatures
```

Comment:
Is that a public api (that is used for example by services / EVM / ...) or is that aprivate api within the platform?

Reply:
I've clarified that the app asks the platform to sign a message.